### PR TITLE
core(redirects): use lantern to compute savings

### PR DESCRIPTION
**Pull request by patrickhulce**
should type-check pending #5072 🌮 

mostly whitespace changes, does the exact same thing as before but using the lantern graphs `startTime` instead of observed

### CODE REVIEW

File: lighthouse-core/audits/redirects.js [hunk 1]
```diff
@@ -20,68 +20,92 @@ class Redirects extends Audit {
...
+    const devtoolsLog = artifacts.devtoolsLogs[Audit.DEFAULT_PASS];
+    const simulatorData = {devtoolsLog, settings: context.settings};
+    const simulator = await artifacts.requestLoadSimulator(simulatorData);
+
+    // TODO(phulce): split MetricComputationInput type and avoid this cast
+    const metricSettings = /** @type {LH.Config.Settings} */ ({throttlingMethod: 'simulate'});
+    const metricComputationData = {trace, devtoolsLog, simulator, settings: metricSettings};
+
+    const mainResource = await artifacts.requestMainResource({URL: artifacts.URL, devtoolsLog});
+    const metricResult = /** @type {LH.Artifacts.LanternMetric} */ (
```
Comment (brendankenny) [inline lighthouse-core/audits/redirects.js]: any reason to not just ask for `artifacts.requestLanternInteractive()` directly?

File: lighthouse-core/audits/redirects.js [hunk 2]
```diff
@@ -20,68 +20,92 @@ class Redirects extends Audit {
...
+      });
+    }
+
+    for (let i = 1; i < redirectRequests.length; i++) {
+      const initialRequest = redirectRequests[i - 1];
+      const redirectedRequest = redirectRequests[i];
+
+      const initialTiming = nodeTimingsByUrl.get(initialRequest.url);
+      const redirectedTiming = nodeTimingsByUrl.get(redirectedRequest.url);
+      if (!initialTiming || !redirectedTiming) continue;
```
Comment (brendankenny) [inline lighthouse-core/audits/redirects.js]: wouldn't this be a error with the graph if any of the redirects aren't in there? Maybe we should throw or log an error in that (unlikely?) case?

File: lighthouse-core/audits/redirects.js (hunk 1 above)
Comment (patrickhulce) [reply lighthouse-core/audits/redirects.js]: yeah I suppose at this point it's not that different 👍 done

File: lighthouse-core/audits/redirects.js (hunk 2 above)
Comment (patrickhulce) [reply lighthouse-core/audits/redirects.js]: fair, I'll switch to error

**Pull request merged**